### ISSUE TITLE
RStudio desktop hangs at start-up in several recent operating system versions e.g. Ubuntu 22.04 & Fedora 35

### ISSUE THREAD

**Issue by fthx** [rstudio/rstudio#9854]
- [x ] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x ] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x ] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by ronblum** [comment]
@fthx Thank you for raising this!

1) RStudio only supports the LTS releases, which in the case of Ubuntu, the latest is 20.04.x. The next major version will be 22.04.
2) That said: Is there any output to the terminal?
3) Was it working in a previous version of Ubuntu, LTS or otherwise?
4) Are you using Wayland, and if so, does switching away from it resolve the issue? See "Xorg will still be there" at https://www.omgubuntu.co.uk/2021/01/ubuntu-21-04-will-use-wayland-by-default

**Comment by fthx** [comment]
Thanks for answering this.

Yes I know that LTS only is supported but I got very new hardware so i'll go for 22.04 later, starting with 21.10 now. "That said" :-) I did use RStudio on non-LTS and that's the first time I experience such a lock. First reaction, I did switch to Xorg but same problem here. It was working on my previous 20.04 and more recently in my Debian 11 setup. (Note that Debian 11 / Ubuntu 20.04 are not listed in RStudio Downloads!)
In terminal, same as usual, as far as I can remember:
`Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.`

**Comment by fthx** [comment]
I'm trying to install RS on my Ubuntu 21.10 VM in Debian 11.

**Comment by fthx** [comment]
Same issue.

**Comment by N0rbert** [comment]
Reproducible way on latest *Ubuntu-MATE 21.10 "Impish Indri" - Alpha amd64 (20210919)* with latest RStudio 1.4.1717:

```
cd ~/Downloads
wget -c https://rstudio.org/download/latest/stable/desktop/bionic/rstudio-latest-amd64.deb
sudo apt-get install -y r-base-dev ./rstudio-latest-amd64.deb
```

Then open two terminals - first to launch RStudio, second - to monitor the logs by `sudo tail -f ~/.xsession-errors /var/log/*`.
In first terminal run `rstudio` command.
In the second terminal get the following:

```
==> /var/log/apport.log <==
ERROR: apport (pid 6828) Sun Sep 19 13:42:37 2021: called for pid 6827, signal 31, core limit 0, dump mode 1
ERROR: apport (pid 6828) Sun Sep 19 13:42:37 2021: executable: /usr/lib/rstudio/libexec/QtWebEngineProcess (command line "/usr/lib/rstudio/libexec/QtWebEngineProcess\ --type=renderer\ --disable-gpu-memory-buffer-video-frames\ --enable-threaded-compositing\ --enable-viewport\ --use-gl=desktop\ --enable-features=AllowContentInitiatedDataUrlNavigations\ --disable-features=MojoVideoCapture,SurfaceSynchronization,UseModernMediaControls,UseVideoCaptureApiForDevToolsSnapshots\ --disable-databases\ --service-pipe-token=972834821355294252\ --lang=en-US\ --webengine-schemes=qrc:sLV\ --num-raster-threads=1\ --service-request-channel-token=972834821355294252\ --renderer-client-id=3\ --shared-files")
ERROR: apport (pid 6828) Sun Sep 19 13:42:37 2021: is_closing_session(): no DBUS_SESSION_BUS_ADDRESS in environment
ERROR: apport (pid 6828) Sun Sep 19 13:42:37 2021: wrote report /var/crash/_usr_lib_rstudio_libexec_QtWebEngineProcess.1000.crash
ERROR: apport (pid 6841) Sun Sep 19 13:42:38 2021: called for pid 6840, signal 31, core limit 0, dump mode 1
ERROR: apport (pid 6841) Sun Sep 19 13:42:38 2021: executable: /usr/lib/rstudio/libexec/QtWebEngineProcess (command line "/usr/lib/rstudio/libexec/QtWebEngineProcess\ --type=renderer\ --disable-gpu-memory-buffer-video-frames\ --enable-threaded-compositing\ --enable-viewport\ --use-gl=desktop\ --enable-features=AllowContentInitiatedDataUrlNavigations\ --disable-features=MojoVideoCapture,SurfaceSynchronization,UseModernMediaControls,UseVideoCaptureApiForDevToolsSnapshots\ --disable-databases\ --service-pipe-token=6814617557611600180\ --lang=en-US\ --webengine-schemes=qrc:sLV\ --num-raster-threads=1\ --service-request-channel-token=6814617557611600180\ --renderer-client-id=3\ --shared-files")
ERROR: apport (pid 6841) Sun Sep 19 13:42:38 2021: is_closing_session(): no DBUS_SESSION_BUS_ADDRESS in environment
ERROR: apport (pid 6841) Sun Sep 19 13:42:38 2021: apport: report /var/crash/_usr_lib_rstudio_libexec_QtWebEngineProcess.1000.crash already exists and unseen, doing nothing to avoid disk usage DoS
ERROR: apport (pid 6945) Sun Sep 19 13:43:05 2021: called for pid 6944, signal 31, core limit 0, dump mode 1
ERROR: apport (pid 6945) Sun Sep 19 13:43:05 2021: executable: /usr/lib/rstudio/libexec/QtWebEngineProcess (command line "/usr/lib/rstudio/libexec/QtWebEngineProcess\ --type=renderer\ --disable-gpu-memory-buffer-video-frames\ --enable-threaded-compositing\ --enable-viewport\ --use-gl=desktop\ --enable-features=AllowContentInitiatedDataUrlNavigations\ --disable-features=MojoVideoCapture,SurfaceSynchronization,UseModernMediaControls,UseVideoCaptureApiForDevToolsSnapshots\ --disable-databases\ --service-pipe-token=2602721841825081561\ --lang=en-US\ --webengine-schemes=qrc:sLV\ --num-raster-threads=1\ --service-request-channel-token=2602721841825081561\ --renderer-client-id=3\ --shared-files")
ERROR: apport (pid 6945) Sun Sep 19 13:43:05 2021: is_closing_session(): no DBUS_SESSION_BUS_ADDRESS in environment
ERROR: apport (pid 6945) Sun Sep 19 13:43:05 2021: apport: report /var/crash/_usr_lib_rstudio_libexec_QtWebEngineProcess.1000.crash already exists and unseen, doing nothing to avoid disk usage DoS
ERROR: apport (pid 6957) Sun Sep 19 13:43:06 2021: called for pid 6956, signal 31, core limit 0, dump mode 1
ERROR: apport (pid 6957) Sun Sep 19 13:43:06 2021: executable: /usr/lib/rstudio/libexec/QtWebEngineProcess (command line "/usr/lib/rstudio/libexec/QtWebEngineProcess\ --type=renderer\ --disable-gpu-memory-buffer-video-frames\ --enable-threaded-compositing\ --enable-viewport\ --use-gl=desktop\ --enable-features=AllowContentInitiatedDataUrlNavigations\ --disable-features=MojoVideoCapture,SurfaceSynchronization,UseModernMediaControls,UseVideoCaptureApiForDevToolsSnapshots\ --disable-databases\ --service-pipe-token=4596915961412917752\ --lang=en-US\ --webengine-schemes=qrc:sLV\ --num-raster-threads=1\ --service-request-channel-token=4596915961412917752\ --renderer-client-id=3\ --shared-files")
ERROR: apport (pid 6957) Sun Sep 19 13:43:06 2021: is_closing_session(): no DBUS_SESSION_BUS_ADDRESS in environment
ERROR: apport (pid 6957) Sun Sep 19 13:43:06 2021: apport: report /var/crash/_usr_lib_rstudio_libexec_QtWebEngineProcess.1000.crash already exists and unseen, doing nothing to avoid disk usage DoS
ERROR: apport (pid 7009) Sun Sep 19 13:43:24 2021: called for pid 7005, signal 6, core limit 18446744073709551615, dump mode 1
ERROR: apport (pid 7009) Sun Sep 19 13:43:24 2021: ignoring implausibly big core limit, treating as unlimited
ERROR: apport (pid 7009) Sun Sep 19 13:43:24 2021: executable: /usr/bin/gdb (command line "/usr/bin/gdb --ex file\ "/usr/lib/rstudio/libexec/QtWebEngineProcess" --ex core-file\ /tmp/apport_core_8a5f26qo --batch --ex set\ backtrace\ limit\ 2000 --ex p\ -99 --ex info\ registers --ex p\ -99 --ex x/16i\ $pc --ex p\ -99 --ex bt\ full --ex p\ -99 --ex thread\ apply\ all\ bt\ full --ex p\ -99 --ex print\ __abort_msg->msg --ex p\ -99 --ex print\ __glib_assert_msg --ex p\ -99 --ex print\ (char*)\ __nih_abort_msg --ex p\ -99")
ERROR: apport (pid 7009) Sun Sep 19 13:43:24 2021: debug: session gdbus call: (true,)

ERROR: apport (pid 7009) Sun Sep 19 13:43:28 2021: wrote report /var/crash/_usr_bin_gdb.1000.crash
ERROR: apport (pid 7009) Sun Sep 19 13:43:28 2021: writing core dump to core (limit: -1)
ERROR: apport (pid 7009) Sun Sep 19 13:43:37 2021: writing core dump core of size 114429952
```

Apport crash dumps are in [apport.zip](https://github.com/rstudio/rstudio/files/7191940/apport.zip)

RStudio is not functional, its window looks like shown below:

>![rstudio](https://user-images.githubusercontent.com/309456/133924837-2ef33337-ab37-4143-acaa-657ad6b94ad5.png)

Installing preview version 2021.09.0-preview+341 does not help.



Tested this on old-but-good Lenovo SL500 with integrated Intel graphics using default video-driver. 
Please note that the same RStudio 1.4.1717 version works normally on Ubuntu MATE 21.04 (non-LTS, but ok for some users) here with the same hardware.

**Comment by ronblum** [comment]
Thank you both, including comparing hardware! I'm tagging this, especially as issues with Ubuntu 21.10 could potentially arise in 22.04 (LTS) when it's released.

**Comment by wagafo** [comment]
I confirm it with Ubuntu 21.10. It would be nice to get a fix before Ubuntu 22.04. This is the first time that RStudio cannot be used with a Non-LTS version, as far as I know. I've been updating to the 6 month version for years and never had a problem with RStudio.

**Comment by fthx** [comment]
Ok guys, I got it.
I was reading some preliminary Impish release notes and there was:
https://bugs.launchpad.net/ubuntu/+source/glibc/+bug/1944468

So:
`rstudio --no-sandbox`
does make RS start, before get it fixed.

**Comment by wagafo** [comment]
Thanks so much, it works.

**Comment by ronblum** [comment]
@fthx thank you for doing the research and figuring out a workaround, and @wagafo for confirming that the workaround works! Having it here will be very helpful in the meantime for others, too, so we truly appreciate your help.

**Comment by fthx** [comment]
Should be closed soon.
:-)
https://lists.ubuntu.com/archives/impish-changes/2021-September/006481.html

**Comment by fthx** [comment]
I installed new glibc from proposed and RStudio starts.

**Comment by gsgxnet** [comment]
Just for information:
the same problem still persists on non-supported OpenSUSE Tumbleweed, see additional replies at 
https://community.rstudio.com/t/rstudio-does-not-start-in-ubuntu-21-10/115852/5
One user there reports also that the Ubuntu issue is fixed for him with the `glibc 2.34-0ubuntu3version`

**Comment by mikebessuille** [comment]
Reopened; we need to investigate this as it affects several OS versions we currently don't support (but some of which we will eventually support).  
Graham suggests upgrading to a new version of Boost may resolve?
see: https://rstudio.slack.com/archives/C0DAAVDJQ/p1636477205095100

**Comment by jmcphers** [comment]
Another report of this on Fedora 35: https://community.rstudio.com/t/rstudio-fails-to-launch-r-session-at-startup-in-fedora/121667

**Comment by skeydan** [comment]
Interesting, it wouldn't have occurred to me to try with sudo ... 
Seems like I need to urgently get that update ... :-)

**Comment by gsgxnet** [comment]
good you reopened that issue. So far the best option under Linux (and probably MacOS as well) is to 
`export RSTUDIO_CHROMIUM_ARGUMENTS="--no-sandbox"`  
somehow into your shell environment, e.g. from `.bashrc`

**Comment by skeydan** [comment]
@jmcphers the kernel update to 5.14.17-301 indeed fixed this for me.

**Comment by englianhu** [comment]
These links workable to me when startup blank page.

- Continuing the discussion from [GitHub SSH doesn&#x27;t work](https://community.rstudio.com/t/github-ssh-doesnt-work/124418/2):
- Continuing the discussion from [Linux 🐧 RStudio Startup Blank Page](https://community.rstudio.com/t/linux-rstudio-startup-blank-page/124168/2):
- https://github.com/STAT545-UBC/Discussion/issues/93#issuecomment-391606351

**Comment by veroandreo** [comment]
I have updated my laptop to Fedora 35 today (so, all packages are up to date) and I'm experiencing the same issue. Adding the `--no-sandbox` flag in the GUI icon command or in the terminal allows me to start RStudio, but it all feels extra slow and clunky. Getting RStudio info (Help -> About RStudio) took forever and made the whole thing to hang, unresponsive. I cannot do anything else within RStudio, except for closing it. 

Any other suggestion or updates on this matter?

**Comment by gsgxnet** [comment]
Since November I do use the 
`export RSTUDIO_CHROMIUM_ARGUMENTS="--no-sandbox"`
trick. 
From my point of view, the `glibc` in version 2.34 is the major cause of trouble.  
At this time my system uses `glibc-2.34-4.3.x86_64`. Still the same problem withRStudio (2022.02.0-daily+337 "Prairie Trillium" Daily ...for OpenSUSE 15 ...) .
will not start when the `--no-sandbox` option is not set.
Sad.
At the moment there is no newer `glibc` in the update pipeline for my Linux. 
May be I will give a newer RStudio Daily a try in the next few days. 
Will report then.

**Comment by gsgxnet** [comment]
Same story with the newest preview from the http://dailies.rstudio.com:  
RStudio 2022.02.0-preview+392 "Prairie Trillium" Preview (9b625ecd4777b148ffb1c322c2184e18bbf49b0c, 2022-01-21) for OpenSUSE 15
Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) QtWebEngine/5.12.8 Chrome/69.0.3497.128 Safari/537.36  
No go on OpenSUSE Tumbleweed when started without the `--no-sandbox`option.  
I am still not sure whether this is a `glibc`or RStudio fault.

**Comment by fthx** [comment]
Got this bug again!
Ubuntu 22.04 dev
with:
[RStudio 2022.02.0+423 - Ubuntu 18/Debian 10 (64-bit)](https://s3.amazonaws.com/rstudio-ide-build/desktop/bionic/amd64/rstudio-2022.02.0-423-amd64.deb)

**Comment by ronblum** [comment]
@fthx Thank you for the update! We're targeting for the next release of RStudio, Spotted Wakerobin 2022.06 which is scheduled for June (date subject to change). We already have daily builds for it, but it's early in the development process and we haven't addressed this yet.

Note for `qa repro`: If the timing allows it, we can work to reproduce this on the current development or upcoming beta release of Ubuntu 22.04 Jammy Jellyfish. If not, the final release is tentatively April 21. See https://discourse.ubuntu.com/t/jammy-jellyfish-release-schedule/ for planned dates.

**Comment by sjvrensburg** [comment]
Please note that this bug also affects EndeavourOS and (probably) other Arch Linux derived distributions. See this post on the RStudio community forums:
[https://community.rstudio.com/t/rstudio-startup-issues/129247?u=stefanj](https://community.rstudio.com/t/rstudio-startup-issues/129247?u=stefanj)
While I realise that you do not officially support Arch-based distros, the problem appears to be quite pervasive among several different OSs.

**Comment by N0rbert** [comment]
The situation gone worse - stable RStudio version 2021.09.2+382 downloaded as usual from https://rstudio.org/download/latest/stable/desktop/bionic/rstudio-latest-amd64.deb shows empty window on todays *Ubuntu-MATE 22.04 LTS _Jammy Jellyfish_ - Alpha amd64 (20220217)* installation.
Adding `--no-sandbox` helps as "usual".

**Comment by fthx** [comment]
Same with latest v. 2022.02.0+442.

Please note this comment:
https://bugs.launchpad.net/ubuntu/+source/glibc/+bug/1944468/comments/40

**Comment by kevinushey** [comment]
We're investigating whether updating to the latest version of the Qt 5.12.x series on Linux can help resolve this issue.

**Comment by gtritchie** [comment]
No luck there; I did a build with Qt 5.12.12 and the problem remains.

**Comment by nowshed-imran** [comment]
Can confirm the same problem in Archlinux package.
https://aur.archlinux.org/packages/rstudio-desktop-bin

**Comment by gsgxnet** [comment]
Did anybody ever pinpoint the root cause for there hangs? 
Is it Electron, is it glibc, is it Qt, is it RStudio itself?  
I have no idea to find out how I could trace it.

**Comment by ronblum** [comment]
@gsgxnet Not yet—@kevinushey and @gtritchie tried with an updated version of Qt, but no luck.

**Comment by gsgxnet** [comment]
@ronblum, yeah I realized that. It looks more like trial and error. I had been hoping for a systematic option. My past experience has shown me though, sometimes the systematic way is not available, at least not at the beginning. And if so, the way is very rocky and a long run.  
Looks like this case needs patience

**Comment by mwhudson** [comment]
Hi! I'm the glibc maintainer in Ubuntu.

It's not completely 100% certain from the comments in this bug, but the problem is very likely to be a bug in the handling of the clone3 syscall by chromium's security sandbox. clone3 was added to the kernel in 5.3 but glibc did not start calling it until 2.34. I patched the calling code out of our 2.34 packages before the release of 21.10 but restored the code when updating the development series to glibc 2.35, so here we are again. Obviously, we can patch it out again before the release of the LTS but I'd rather not.

As the problem is in chromium, upgrading Qt or Boost is not going to help. The fix from chromium that's needed is https://chromium-review.googlesource.com/c/chromium/src/+/2936184, which is part of Chromium 93.0.4532.0 and newer. This fix is part of the latest electron releases of at least v13 and newer. I started at the RStudio build process for a while but completely failed to understand which version of electron is being used.

Hopefully this helps clarify things.

**Comment by gsgxnet** [comment]
So the right `electron` would fix it. Great. 
I fast flipped (amateur attempt) through the RStudio dailies files and build commits, but could not find the right one where the `electron` version is set.

Hopefully one of the next builds will include the right one.  

And sure there is no long term benefit in a for a workaround modified `glibc`. 

Many thanks.

**Comment by kevinushey** [comment]
Just to confirm, the current release of RStudio consumes Chromium through QtWebEngine. Older versions of QtWebEngine generally include an older version of Chromium, but Qt is pretty diligent about backporting fixes as necessary. However, it seems like they haven't yet backported this fix (at least for the Qt 5.12.x series.)

I saw one bug report for Qt here:

https://bugreports.qt.io/browse/QTBUG-96214?jql=text%20~%20%22glibc%22

so it's possible the fix was backported for Qt 5.15.x, but not Qt 5.12.x. (I haven't investigated further to confirm, though.)

**Comment by mwhudson** [comment]
Ah that would make sense. Unfortunately it looks like Qt 5.12 just went EOL?

**Comment by gsgxnet** [comment]
@kevinushey - yeah that 5.15.5 fix seems the needed one.  Does 5.15 have a different license than 5.12. which would make it more complicated to integrate it into the RStudio Free / OpenSource versions?

**Comment by kevinushey** [comment]
It doesn't; the challenge here is that (in our experience) Qt updates often break different pieces of RStudio functionality, and running down each piece of breakage is a time-consuming task.

**Comment by mikebessuille** [comment]
We are in the process of converting the IDE to Electron (from QTWebEngine); should have a preview of that by June.  It sounds like that will fix the problem as we'll be using one of the latest versions of Electron.  It would be prohibitively costly for us to try to take a new version of QT before that...

**Comment by writerjkf** [comment]
Just had this issue pop up on my Arch Linux machine yesterday. Is there a workaround while we wait for a more permanent fix? Running RStudio as root seems to work, but its a less-than-ideal solution, even if it's temporary.

**Comment by gsgxnet** [comment]
@mikebessuille - understandable. But how to address the problem until then? Would you apply the 5.15.5 patch mentioned by @kevinushey yourself on 5.12.x, license permitting? 
I know myself how to cope with the problem, and a few others as well. But other users on newer Linux variants will either abandon RStudio or file yet another issue report. Not good. 
Would delivering RStudio with the `export RSTUDIO_CHROMIUM_ARGUMENTS="--no-sandbox"` set be an official work around?

**Comment by fthx** [comment]
Ok I've modified my /usr/share/applications/rstudio.desktop to include --no-desktop since I feel it will be a longstanding pain.

**Comment by mwhudson** [comment]
I think it would be fairly easy to port the fix to the version of chromium used by qt 5.12 but the issue I think is that the rstudio downloads a binary build of it? So you'd need to convince whoever maintains the binary builds to apply the patch or build packages yourself. It might be worth talking to the maintainers I guess, as anyone using their qt 5.12 builds with webengine on a new distro is going to hit this.

**Comment by gsgxnet** [comment]
Yeah, looks like there is no alternative to rely on the download (online) version of Qt 5.12. etc. The license states (https://www.qt.io/blog/qt-offering-changes-2020), near top:  
"Long-term-supported (LTS) releases and the offline installer will become available to commercial licensees only" 
and further down again:
"The offline installer will also become commercial-only. We found that this feature adds utility on a company-wide level, which allows us to make commercial licenses more attractive to enterprises without significantly inconveniencing open-source users."

**Comment by gsgxnet** [comment]
@mikebessuille 
> We are in the process of converting the IDE to Electron (from QTWebEngine); should have a preview of that by June. It sounds like that will fix the problem as we'll be using one of the latest versions of Electron. It would be prohibitively costly for us to try to take a new version of QT before that...

I just upgraded to the newest daily build: 
RStudio 2022.06.0-daily+136 "Spotted Wakerobin" Daily (2c92cd9adebc7c18c052d2547948f07db835b13c, 2022-02-21) for OpenSUSE 15
Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) QtWebEngine/5.12.8 Chrome/69.0.3497.128 Safari/537.36  
seeing 2022.06 there, which might mean June edition,  did let me hope for a version on the path away from Qt to Electron. Not so.  Anyhow, I am upgrading about once per week to the most current daily build, so when there is some Electron based version delivered sometime in the future, I will test run it and will report issues if they show up.  
Good luck with the migration.

**Comment by mikebessuille** [comment]
@gsgxnet sorry for the confusion!  The electron builds of the IDE are not yet in preview; that work is being done in parallel to the main development branch; by the June (2022.06) release, we'll still have the old QTwebengine version of the IDE, but we'll also have a preview of an electron version around the same time.  Not long after that, we'll start releasing the IDE only with electron.
Given the attention this issue has, we'll have a look at short-term alternatives.

**Comment by gsgxnet** [comment]
> @gsgxnet sorry for the confusion! The electron builds of the IDE are not yet in preview ...

@mikebessuille sorry me - I had confused my self, assuming your daily builds are not part of the preview line but part of the development line.

**Comment by SeverinBang** [comment]
> Ok I've modified my /usr/share/applications/rstudio.desktop to include --no-sandbox since I feel it will be a longstanding pain.

This fixes the white screen issue when opening a new instance, but sadly the problem persists when opening a new R(studio) session from _within_ an working instance.

**Comment by gsgxnet** [comment]
> > Ok I've modified my `/usr/share/applications/rstudio.desktop` to include `--no-sandbox` since I feel it will be a longstanding pain.
> 
> This fixes the white screen issue when opening a new instance, but sadly the problem persists when opening a new R(studio) session from _within_ an working instance.

@SeverinBang for these see my solution under Linux and MacOS, above: https://github.com/rstudio/rstudio/issues/9854#issuecomment-972033704  

You should set the environment variable 
`export RSTUDIO_CHROMIUM_ARGUMENTS="--no-sandbox"`
somehow into your shell environment, e.g. from `.bashrc`.
Then RStudio at start will follow that option, will not activate the sandbox mode and will not fail with a blank screen. This works for new sessions(instances) started from a running session(instance) too. That way I do not have any issues starting RStudio however I do it.

**Comment by SeverinBang** [comment]
> 

I have tried that, also setting `export RSTUDIO_CHROMIUM_ARGUMENTS="--disable-seccomp-filter-sandbox"` in my .zshrc as discussed in at https://aur.archlinux.org/packages/rstudio-desktop-bin#comment-853431.

I have tried it with the stable, preview and daily branch (each including purging all config files), and all show the same behavior: natively running the program works, open a new instance from an active instance fails.

**Comment by ronblum** [comment]
@SeverinBang Which version of Linux are you using? I tried both options,
- export RSTUDIO_CHROMIUM_ARGUMENTS="--no-sandbox"
- export RSTUDIO_CHROMIUM_ARGUMENTS="--disable-seccomp-filter-sandbox"

in `.bashrc`. RStudio Desktop 2022.02.0+443 Prairie Trillium on Ubuntu 22.04 (dev) with R 4.1.2, with both the original launch and using <kbd>Session -> New Session</kbd>, seems to be working.

**Comment by SeverinBang** [comment]
@ronblum My current setup is:

OS: Manjaro Linux x86_64 
Host: XPS 15 7590 
Kernel: 5.16.11-2-MANJARO 
Shell: zsh 5.8.1 
Resolution: 3840x2160, 3840x2160 
DE: Plasma 5.24.2 
WM: KWin 
CPU: Intel i7-9750H (12) @ 4.500GHz 
GPU: NVIDIA GeForce GTX 1650 Mobile / Max-Q 
GPU: Intel CoffeeLake-H GT2 [UHD Graphics 630] 
Memory: 10316MiB / 15634MiB 

with: r 4.1.2-3
and the Rstudio versions i have tried are:
stable: [2021.09.2.382-1](https://aur.archlinux.org/packages/rstudio-desktop-bin)
preview: [2022.02.0.443-1](https://aur.archlinux.org/packages/rstudio-desktop-preview-bin)
daily: [2022.06.0.daily.152-1](https://aur.archlinux.org/packages/rstudio-desktop-daily-bin)

**Comment by ronblum** [comment]
@SeverinBang Thanks! It's too bad that it's not working. I'm not sure what we can do about it, but hopefully someone else using Manjaro might stumble across this issue and at least see that they're not alone—and maybe even come up with another workaround.

**Comment by SeverinBang** [comment]
addendum: setting the environment variable and editing the .desktop file (as described above) did not help alone, but if i start rstudio from the terminal with `rstudio-bin --no-sandbox` everything works, including opening new sessions from within a session. still weird.

**Comment by davidbudzynski** [comment]
I'm also having the same issues on Arch Linux. I'm using daily-bin version from the AUR and `rstudio-daily-bin no-sandbox` seems to fix the problem. I'd appreciate if this could be fixed altogether!

**Comment by subwaystation** [comment]
Observing the same as @davidbudzynski.

**Comment by chendaniely** [comment]
fwiw: one of the [commenters](https://aur.archlinux.org/packages/rstudio-desktop-daily-bin#comment-853678) from the Arch Linux AUR suggested using `--disable-seccomp-filter-sandbox` instead of `--no-sandbox` because of security risks.

**Comment by cribari** [comment]
I run Arch Linux + KDE + sddm under X11. rstudio-desktop-bin is installed on my system (from the AUR). I am able to start rstudio using sudo, but the other proposed solutions (--no-sandbox and --disable-seccomp-filter-sandbox) do not work for me. The problem started recently. Suggestions are welcome. See comment #2 at 

https://bbs.archlinux.org/viewtopic.php?pid=2027043#p2027043

**Comment by dagdrommer** [comment]
> I run Arch Linux + KDE + sddm under X11. rstudio-desktop-bin is installed on my system (from the AUR). I am able to start rstudio using sudo, but the other proposed solutions (--no-sandbox and --disable-seccomp-filter-sandbox) do not work for me. The problem started recently. Suggestions are welcome. See comment #2 at
> 
> https://bbs.archlinux.org/viewtopic.php?pid=2027043#p2027043

I have the exact same setup as you @cribari. I fixed it by compiling rstudio from source using the non-binary AUR-App: https://aur.archlinux.org/packages/rstudio-desktop (uninstall the old first)

The comments on the page of non-binary AUR-App also don't indicate the problem of this thread, so I think I am not the only one who uses it as a fix. Compiling takes some time (depending on your system, but it worth it imo),

**Comment by mwt** [comment]
> Does not work for me, Ubuntu 22.04 dev too.

Same here. It does not work without `--no-sandbox` but it works just fine with this option. I have had the exact same issue with several different apps.

**Comment by jmcphers** [comment]
Another report from the community forum: https://community.rstudio.com/t/rstudio-gui-is-empty-for-normal-user/135151/2

**Comment by dat-linux** [comment]
Can someone just summarize what/when/how re a fix for Ubuntu 22.04 users? We have to wait until June?

What are the side effects of running it with --no-sandbox ? 

Thanks.

**Comment by mwt** [comment]
> Can someone just summarize what/when/how re a fix for Ubuntu 22.04 users? We have to wait until June?

I just added --no-sandbox to the application shortcut. It works fine for me on Ubuntu 22.04. I don't understand what the libssl issue is. People were saying that they had to sideload this. I didn't do that.

> What are the side effects of running it with --no-sandbox ? 

I haven't noticed any side effects. Many electron applications require this workaround. I have it set on basically all of them.

**Comment by sscansan** [comment]
Here with Pop_OS! v.22.04, same story with a white not usable window.
I've put  `export RSTUDIO_CHROMIUM_ARGUMENTS="--no-sandbox"` as suggested in `.zshrc` but there are no improvement.
The only way to make it work is to directly launch it from terminal with 
`rstudio --no-sandbox` 
weird.

**Comment by rmolina** [comment]
@sscansan: I'm using Pop_OS! (22.04) too. I added `export RSTUDIO_CHROMIUM_ARGUMENTS="--no-sandbox"` to my `.bashrc`. Launched it from a fresh term and it's working now.

**Comment by juliendiot42** [comment]
@sscansan, I have the same trouble in Pop_Os! (22.04),
A more convenient way to work around than using the terminal can be to create a new Keyboard Shortcuts that launch Rstudio. (Settings -> Keyboard -> View and customise Shortcuts  -> Custom Shortcuts) 
I used `Super + R` for the shortcuts, and `rstudio --no-sandbox` for the command.

**Comment by mwt** [comment]
> A more convenient way to work around than using the terminal can be to create a new Keyboard Shortcuts that launch Rstudio. (Settings -> Keyboard -> View and customise Shortcuts  -> Custom Shortcuts) 

Just edit the .desktop file in `/usr/share/applications/` or copy it to `~/.local/share/applications/` and edit it there.

The second is usually preferred.

**Comment by sscansan** [comment]
> > A more convenient way to work around than using the terminal can be to create a new Keyboard Shortcuts that launch Rstudio. (Settings -> Keyboard -> View and customise Shortcuts  -> Custom Shortcuts) 
> 
> Just edit the .desktop file in `/usr/share/applications/` or copy it to `~/.local/share/applications/` and edit it there.
> 
> The second is usually preferred.
> 
> 

I have done exactly this and solved the issue. Thank you!

The question I have is: how big is the threat deactivating sandbox in RStudio? I guess has implications in the security of the whole system. Will the RStudio team work to solve this? 
Regards

**Comment by dat-linux** [comment]
> @sscansan: I'm using Pop_OS! (22.04) too. I added `export RSTUDIO_CHROMIUM_ARGUMENTS="--no-sandbox"` to my `.bashrc`. Launched it from a fresh term and it's working now.

You're better of putting this in **`.profile`**, otherwise it's only available from a bash terminal.

**Comment by mikebessuille** [comment]
We are working to solve this in the electron version of the desktop, which will be in preview shortly (by June, if not earlier).    We are considering launching with --no-sandbox by default in the current QT version of desktop, when on Ubuntu, but are first reviewing the security concerns.

**Comment by dat-linux** [comment]
> We are working to solve this in the electron version of the desktop, which will be in preview shortly (by June, if not earlier). We are considering launching with --no-sandbox by default in the current QT version of desktop, when on Ubuntu, but are first reviewing the security concerns.

How can I get a hold of the QT desktop version or RStudio? Is there an official release or is this in testing? Is the QT version just a native wrapper around electron, or is it purely native?

**Comment by gtritchie** [comment]
RStudio has always been built on Qt. The Electron version is new and still under development and will eventually replace the Qt version.

So, when you download the released version of RStudio or the daily builds, those are using Qt.

**Comment by gtritchie** [comment]
Further clarification: most of the RStudio user interface is HTML/JavaScript-based, that has always been the case. Currently that is rendered by RStudio Desktop using QtWebEngine, but we are in the process of moving it to Electron.

**Comment by dat-linux** [comment]
> RStudio has always been built on Qt. The Electron version is new and still under development and will eventually replace the Qt version.
> 
> So, when you download the released version of RStudio or the daily builds, those are using Qt.

Thanks.

Not a fan of electron, they're basically desktop apps in a browser, ugh.

**Comment by gtritchie** [comment]
> Not a fan of electron, they're basically desktop apps in a browser, ugh. 

RStudio has always been a web app. It was an "Electron" app (conceptually) long before Electron came along.

**Comment by kevinushey** [comment]
Sorry again for the delay. Daily builds of RStudio built for Ubuntu Jammy are now available on the daily builds page:

https://dailies.rstudio.com/

The current daily build is https://s3.amazonaws.com/rstudio-ide-build/desktop/jammy/amd64/rstudio-2022.06.0-daily-340-amd64.deb, and we'd appreciate if you could test and report if RStudio is now able to launch without issue for you.

The next builds of RStudio for other platforms should also support newer versions of glibc.

**Comment by michkam89** [comment]
Hi @kevinushey, just FYI, I confirm that this build fixed the issue on my machine. Thanks!

**Comment by ronblum** [comment]
@michkam89 What operating system are you using? Just so I can track where we know for certain that it works.

---

Verified that the following install and run on Ubuntu 22.04 with R 4.1.2:
- RStudio Desktop 2022.06.0-daily+340
- RStudio Server 2022.06.0-daily+340

Will test on Fedora next.

Please note that the dailies are intended for development and testing purposes.

**Comment by dalewsteele** [comment]
Verified on Ubuntu 22.04 (upgraded from 21.10) with R 4.2

**Comment by samuelkn** [comment]
Verified on Manjaro (Kernel 5.17) and R 4.1.3 (--no sandbox issue verified on  2022.02.1.461)
Tested RStudio Desktop 2022.06.0-daily-341 and working fine.
Also already working with daily built 209.

**Comment by DanKHansen** [comment]
Confirming Rstudio 2022.06.0 Build 340re-release build: 2022.06.0 Build 340 is working on Pop!_OS 22.04 LTS

**Comment by WillTheGeek** [comment]
Confirming rstudio-2022.06.0-daily-341-amd64.deb works on Ubuntu 22.04 with R 4.1.2

**Comment by kdl0013** [comment]
> @sscansan: I'm using Pop_OS! (22.04) too. I added `export RSTUDIO_CHROMIUM_ARGUMENTS="--no-sandbox"` to my `.bashrc`. Launched it from a fresh term and it's working now.

I am also using Pop_OS! (22.04) and can confirm that this solution works.

**Comment by pinduzera** [comment]
Pop_OS! (22.04) with R 4.2 works as intended using 2022.06.0-daily+343

**Comment by gsgxnet** [comment]
openSUSE Tumbleweed (rolling release, refreshed at April 30th) works generally as intended using `2022.06.0-daily+343` electron without having to set in your environment `export RSTUDIO_CHROMIUM_ARGUMENTS="--no-sandbox".

One issue remains (or is new?) - when starting from bash by `/usr/bin/rstudio &` I can see these warnings: 
```
[20765:0501/105924.463898:ERROR:gpu_init.cc(446)] Passthrough is not supported, GL is desktop, ANGLE is 
[20765:0501/105924.482149:ERROR:sandbox_linux.cc(377)] InitializeSandbox() called with multiple threads in process gpu-process.
```
This might be caused by the special circumstance of my setup, a dual GPU Laptop, one Intel embedded, one NVidia.

**Comment by ciremem** [comment]
> 2022.06.0-daily+343

Thank you so much

**Comment by ronblum** [comment]
Thank you all for verifying this! The info you're providing is very good to know.

@gsgxnet Thanks for checking on openSUSE! The `Passthrough` is noted in issue #10782. The `InitializeSandbox` is a new one, though. Would you be willing to [open a new issue](https://github.com/rstudio/rstudio/issues/new?assignees=&labels=&template=bug_report.md) for it?

**Comment by macramole** [comment]
2022.06.0 Build 340 works where on Pop Os 22.04 with R version 4.1.2

thanks !

**Comment by fthx** [comment]
build 348 electron works well (except copy-paste between different tabs, maybe inside a tab too).

**Comment by ronblum** [comment]
@fthx I'm actually writing up a bug for the copy-paste issue right now, but thank you for nothing it!

**Comment by gsgxnet** [comment]
> Thank you all for verifying this! The info you're providing is very good to know.
> 
> @gsgxnet Thanks for checking on openSUSE! The `Passthrough` is noted in issue #10782. The `InitializeSandbox` is a new one, though. Would you be willing to [open a new issue](https://github.com/rstudio/rstudio/issues/new?assignees=&labels=&template=bug_report.md) for it?

did so, see #11114

**Comment by fthx** [comment]
alt + - shortcut does not work in electron

**Comment by gsgxnet** [comment]
@ronblum  - I wonder would it be a good idea to display a warning at RStudio startup in the electron version if the user started it with the `RSTUDIO_CHROMIUM_ARGUMENTS="--no-sandbox"` setting? 
`--no-sandbox` was a temporary measure to cope with the problems, but in the long run it might become a security issue if kept.

**Comment by fthx** [comment]
Could be useful to have a RStudio repository rather than manually updating.
I'm sure this request has been done before.
:-)

**Comment by mwt** [comment]
> Could be useful to have a RStudio repository rather than manually updating.
> I'm sure this request has been done before.
> :-)
> 

https://mattwthomas.com/mirrors/#rstudio

With a GitHub project [here](https://github.com/mwt/rstudio-apt-repo).

**Comment by NotesOfReality** [comment]
According to my limited understanding RStudio daily builds for Ubuntu 22 run as if `--no-sandbox` is enabled by default (surely not passed by a `.desktop` launcher nor  `.profile`), don't they?
Sorry if my question might very well turn out to be "blindly stupid".

**Comment by Eyayaw** [comment]
> Sorry again for the delay. Daily builds of RStudio built for Ubuntu Jammy are now available on the daily builds page:
> 
> https://dailies.rstudio.com/
> 
> The current daily build for Ubuntu Jammy is available at https://s3.amazonaws.com/rstudio-ide-build/desktop/jammy/amd64/rstudio-2022.06.0-daily-340-amd64.deb, and we'd appreciate if you could test and report if RStudio is now able to launch without issue for you.
> 
> The next builds of RStudio for other platforms should also support newer versions of glibc.

Ok, this one was a life saver. Thank you. 
First, `rstudio-2022.02.2-485-amd64.deb` depended on `libssl1*` and it's not easy to install it on Ubuntu 22.04. Once that was solved, rstudio does not open unless with `rstudio --no-sandbox`.  The dailies now solve my problem. Rstudio opens up correctly with R 4.2.

**Comment by ronblum** [comment]
Please note that the dailies are intended for development and testing purposes.

Patterned after from https://github.com/rstudio/rstudio/issues/10902#issuecomment-1121401693

---

### Verified Fixed
    RStudio Edition    : Desktop / Desktop Pro / Server
    RStudio Version(s) : 2022.02.3-daily+489 & 2022.06.0-daily+374
    OS Version         : Ubuntu 22.04 (Desktop & Server), Fedora 35 (RStudio Desktop only)

### Environment details

Does not include RStudio Workbench (see https://github.com/rstudio/launcher/issues/341) or Fedora (https://github.com/rstudio/rstudio/issues/11140).

### Test scenario(s) 

Checked basic functionality, such as startup, running an R script, and knitting an R Markdown file.

For RStudio Desktop Pro, verified both locally and using a remote server.

### Automated tests 

If the bug is present, existing RStudio Server automated test will fail because the application won't run.


Notes: 
1. RStudio Workbench support is covered in private repos at https://github.com/rstudio/rstudio-pro/issues/3422 and https://github.com/rstudio/launcher/issues/341
2. For Fedora 35, only verified RStudio Desktop, not Server/Workbench. If we need server, I'll open up a separate issue.

**Comment by prdm0** [comment]
On Arch Linux and derived distributions the problem continues. A good source of comments on how to resolve it is found [**here**](https://aur.archlinux.org/packages/rstudio-desktop-bin). So far, they're using workarounds, like running `rstudio-bin --no-sandbox` or making changes to the rstudio.desktop file.

**Comment by ronblum** [comment]
@prdm0 Do you mean that the problem still exists in 2022.02.3-daily+489 or 2022.06.0-daily+374? Though keep in mind that these are meant for development and testing right now.

**Comment by phancanhtrinh** [comment]
I still have a problem with an updated version. I cannot install it on ubuntu 22.04

`sudo dpkg -i rstudio-2022.06.0-daily-388-amd64.deb 
`
(Reading database ... 219615 files and directories currently installed.)
Preparing to unpack rstudio-2022.06.0-daily-388-amd64.deb ...
Unpacking rstudio (2022.06.0-daily+388) over (2022.06.0-daily+388) ...
dpkg: dependency problems prevent configuration of rstudio:
 rstudio depends on libssl-dev; however:
  Package libssl-dev is not installed.
 rstudio depends on libclang-dev; however:
  Package libclang-dev is not configured yet.
 rstudio depends on libpq5; however:
  Package libpq5 is not installed.

dpkg: error processing package rstudio (--install):
 dependency problems - leaving unconfigured
Processing triggers for mailcap (3.70+nmu1ubuntu1) ...
Processing triggers for gnome-menus (3.36.0-1ubuntu3) ...
Processing triggers for desktop-file-utils (0.26-1ubuntu3) ...
Processing triggers for hicolor-icon-theme (0.17-2) ...
Processing triggers for shared-mime-info (2.1-2) ...
Errors were encountered while processing:
 rstudio

**Comment by fthx** [comment]
Prefer `sudo apt install ./<deb package>` over dpkg so depends are installed too.

**Comment by phancanhtrinh** [comment]
I have problems too.  I am too tired. Does anyone know how to downgrade to Ubuntu 20.04? I suspect this situation will occur with other packages, not only Rstudio.

`sudo apt install ./rstudio-2022.06.0-daily-388-amd64.deb
`
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
You might want to run 'apt --fix-broken install' to correct these.
The following packages have unmet dependencies:
 libclang-11-dev : Depends: libstdc++-9-dev but it is not installed
                   Depends: libgcc-9-dev but it is not installed
                   Depends: libobjc-9-dev but it is not installed
                   Depends: libclang1-11 (= 1:11.0.0-2~ubuntu20.04.1) but it is not installed
                   Depends: libclang-common-11-dev (= 1:11.0.0-2~ubuntu20.04.1) but it is not installed
 libclang-14-dev : Depends: libobjc-11-dev but it is not installed
                   Depends: libclang1-14 (= 1:14.0.3-1) but it is not installed
                   Depends: libclang-common-14-dev (= 1:14.0.3-1) but it is not installed
 libclang-4.0-dev : Depends: libstdc++-7-dev but it is not installable
                    Depends: libgcc-7-dev but it is not installable
                    Depends: libobjc-7-dev but it is not installable
                    Depends: libclang1-4.0 (= 1:4.0.1-10) but it is not installable
                    Depends: libclang-common-4.0-dev (= 1:4.0.1-10) but it is not installable
 libclang-dev : Depends: libclang-10-dev (>= 10~) but it is not installable
 rstudio : Depends: libssl-dev but it is not installed
           Depends: libpq5 but it is not installed
E: Unmet dependencies. Try 'apt --fix-broken install' with no packages (or specify a solution).

**Comment by kevinushey** [comment]
The binaries built for older versions of Ubuntu won't work with Ubuntu 22.04. Try downloading one of the latest binaries available from here:

https://dailies.rstudio.com/rstudio/spotted-wakerobin/desktop/jammy/

and installing it with `sudo apt install ./rstudio-*`.

**Comment by dat-linux** [comment]
So is there any chance this fix can be folded in to the current version (2022.02.02-*) as a patch release?

**Comment by mikebessuille** [comment]
Yes it's already been backported to Prairie Trillium (2022.02.3), which has not been released yet but you can try it here:
https://dailies.rstudio.com/rstudio/prairie-trillium/

**Comment by dat-linux** [comment]
> @dat-linux Yes it's already been backported to Prairie Trillium (2022.02.3), which has not been released yet but you can try it here: https://dailies.rstudio.com/rstudio/prairie-trillium/

Thanks, but I cannot rely on dailies. When is the release date for stable?

**Comment by mikebessuille** [comment]
@dat-linux I expect in ~ 1-2 months, but not likely in the next 3-4 weeks
